### PR TITLE
Time for a new version - v1.5.0

### LINKED ISSUE #349: Validation data aggregates into training data for multi-drug models
I am not fully sure about the consequences of this bug yet (still running the benchmark with a fixed setup). Still I already wanted to share this:

My setup (the DrugANVAEModel is a custom model I am developing, but this does not matter too much here):

```python
from drevalpy.experiment import drug_response_experiment
from drevalpy.visualization.create_report import create_report
from drevalpy.models import MODEL_FACTORY
from drevalpy.datasets import AVAILABLE_DATASETS

import torch

torch.set_float32_matmul_precision('medium')

dataset = "CTRPv2"
run_id = "drugANVI"
test_mode = "LCO"
n_cv_splits = 5
hyperparameter_tuning = False
path_data = "data"
path_out = "results"

baselines = [
    'NaivePredictor',
    'NaiveTissueMeanPredictor',
    'NaiveTissueDrugMeanPredictor',
]

models = [
    'SimpleNeuralNetwork',
    'ElasticNetModel',
    'PCAElasticNet',
    'ScVIElasticNet',
    'RandomForest',
    'PCARandomForest',
    'ScVIRandomForest',
    'DrugANVAEModel'
]

drug_response_experiment(
            models=[MODEL_FACTORY[model] for model in models],
            baselines=[MODEL_FACTORY[baseline] for baseline in baselines],
            response_data=AVAILABLE_DATASETS[dataset](path_data="data"),
            n_cv_splits=n_cv_splits,
            test_mode=test_mode,
            run_id=run_id,
            path_data=path_data,
            path_out=path_out,
            hyperparameter_tuning=hyperparameter_tuning,
            wandb_project=run_id
)

create_report(run_id=run_id, dataset=dataset)
```

What I did:

1. Run the experiment 'from scratch'
2. Delete the DrugANVAEModel from the results directory
3. Re-run the script (all models except DrugANVAEModel are cached)

What I observe:

During the initial training (from scratch), the DrugANVAEModel trains for 500+ steps per epoch, in the second training it trains only for 300+ steps per epoch. Also the convergence in the initial run is way smoother and reaches a loss of 0.7, whereas in the second run the best I get is 1.7 with very shaky convergence. (I made this observations using weights and biases).

Investigating potential reasons:

It seems like [`get_datasets_from_cv_split()`](https://github.com/daisybio/drevalpy/blob/587cefd8e3c60151e85ee311d2f36d7e7b5a4a47/drevalpy/experiment.py#L1259) returns direct references instead of copies for multi-drug models. This is problematic as [here](https://github.com/daisybio/drevalpy/blob/587cefd8e3c60151e85ee311d2f36d7e7b5a4a47/drevalpy/experiment.py#L224), the validation data is added to the training data. Now when training many models, the validation data is actually added to the training data multiple times, resulting in my observed 500+ steps per epoch compared to 300+ when only one model is actually trained (and the rest is cached).

### PR DESCRIPTION
**Pull request by JudithBernett**
### Changes
* #373 
    * The Sklearn Models now call super().build_model(hyperparameters) before specifying their own additional hyperparameters
    * All models call self.log_hyperparameters for wandb
    * train_final_model: the dataset is now first split into 90/10 train/val, then the best hpams are found, and then, the best model is initialized with the best hpams and train/val are subsetted to the available features. Before, the whole dataset was subsetted to the available features and then split into 90/10. -> now, the model might not train on exactly 90% of the data anymore but as the missing features are expected to be uniformly distributed, this should not matter too much 
* #392 
    *  Random seeding: updated from np.random.seed to np.random.default_rng
    * tests/conftest: session fixtures now resolve the data path from file instead of cwd-relative "../data", so they don't break if a test changes cwd. The pytest_configure os.chdir is kept for now because several other test files still hard-code "../data" string literals

### Bug fixes
* Fixed #349 
* #365 
* #392 
    *  cli_model_testing.randomization_split: fixes https://github.com/daisybio/drevalpy/issues/389. Build the model with every hyperparameter combination and union the resulting randomization test views, since the baseline flexible-inputs refactor (https://github.com/daisybio/drevalpy/pull/373) moved cell_line_views / drug_views from class attributes to instance state populated by build_model. Without this the CLI silently produced no yamls.
    * DrugResponseDataset._cv_splits: was a class-level mutable default (list = []), causing state to leak across instances. Moved initialization into init.
    * noxfile.tests typeguard session: was installing .[fit] which is not a defined extra; switched to .[multiprocessing] (the actual extra) so the session can run
* #409 : first contribution by @jfrog64

### New features
* #348 : thank you @nictru 
* #351 
* New models, updated leaderboard
    * #336  : thank you @nictru 
    * #370 : first contribution by @matvee04 
    * #371 : first contribution by @gretag04 
    * #372  : first contribution by @eleded 
    * #391 : first contribution by @tereshchuk1 -> surpassed by #402 
    * #410 : thank you @tereshchuk1 
* #373 
    * Cell_line_views and drug_views are now treated as hyperparameters in the hyperparameters.yaml file
    * The MultiOmicsRF and MultiOmicsNN have been renamed to MultiFeatureRF and MultiFeatureNN and now also support flexible input: they just concatenate whatever matrices you put in
    * Input can be supplied with a csv and will be read in with the load_generic_csv method. This requires that the csv either has a column CELL_LINE_IDENTIFIER (cell_line_name) or DRUG_IDENTIFIER (pubchem_id), depending on whether a cell_line_view or drug_view was specified. The csv must have the name of the specified view, i.e., you specify "myfeature" -> it requires a "myfeature.csv" file in {data_path}/{dataset_name}/
    * This has caused the Proteomics{RF|ElasticNet}, SingleDrugProteomics{RF|ElasticNet}, and ChemBERTaNN to be redundant. They are now just input specifications of the RF, ElasticNet, SingleDrug{RF|ElasticNet} and SimpleNN
    * If no cell line view / drug view is specified, the IDs will be loaded
    * Sklearn models: If proteomics is specified, the proteomics-specific normalizer will be enabled. If gene expression is specified, a standard scaler will scale it.
    * MultiFeatureRF: If methylation is specified, a PCA will reduce it.
* #411 : thank you @nictru 

### Maintenance
* Added Zenodo badge #352 
* Version updates
* Updated documentation:
    * Added information on the flexible input
    * Added information about weights & biases integration
* .github/workflows/python-package.yml: removed. It duplicated the package build already covered by build_package.yml and ran flake8 with a 127-char line length that conflicted with the 120-char config used by pre-commit in run_tests.yml 
* Paper is finally out! Updated documentation
* #347 
* Updated leaderboard and overview figure #421 

## Version release checklist

Then,

1. Open a PR from development to main with these changes.
2. Wait for a review and merge.
3. Create a new release on GitHub with the version number. Update the release notes with the changes made in this version.
4. If the Docker github action fails (e.g., no space left on device), you can do it manually:
   - Enable Docker buildx locally. If you don't have a builder that supports multi-arch, create one:
   ```{bash}
   docker buildx ls
   docker buildx create --use --name multiarch-builder
   docker buildx inspect --bootstrap
   ```
   - Login to ghcr `docker login --username <GitHub-Username> --password <personal-access-token> ghcr.io`
   - Build the multi-platform image, insert the correct version `docker buildx build --platform linux/amd64,linux/arm64 -t ghcr.io/daisybio/drevalpy:v<INSERT-VERSION> -t ghcr.io/daisybio/drevalpy:latest --push .`

### CODE REVIEW

**Comment by JudithBernett** [comment]
Some formalities still have to be updated: 

* Overview figure for the new baseline models
* Leaderboard for the new baseline models
* Documentation for the new baseline models
* Documentation for the new CLI
* Update contributor list

**Review by PascalIversen — APPROVED** [review]
looks VERY good
